backend/spec/services/spreadsheet_generator_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "roo"
+require "tempfile"
+
+RSpec.describe SpreadsheetGenerator do
+  describe "#perform" do
+    let!(:enrollment) { create(:enrollment, :franceconnect) }
+    let!(:foreign_enrollment) { create(:enrollment, :api_entreprise) }
```

Comment:
```suggestion
    let(:enrollment) { create(:enrollment, :franceconnect) }
    let(:foreign_enrollment) { create(:enrollment, :api_entreprise) }
```

Reply:
Quand je retire le `!` l'enrollment n'est plus initialisé dans les `it block`, cela est probablement du à cause des `context`. 
Vu que l'on ne teste plus le cas d'un fichier sans enrollments, je retire de le context. 

le cas de l'erreur `not_found` du a un fichier sans enrollments sera testé directement dans le spec de la méthode `export`

Reply:
> Quand je retire le ! l'enrollment n'est plus initialisé dans les it block, cela est probablement du à cause des context.

Pourtant il est présent dans `enrollments`, qui est présent dans `spreadsheet_generator`, qui est utilisé dans le `it`
D'ailleurs ici il n'y a pas de notion de foreign enrollment, ce service prend des enrollments et les transforme en xlsx, indépendamment du contexte (et de toute façon ici il n'y a pas de notion de user)

Reply:
je change le nom de foreign enrollment pour en enrollment_2, pour avoir 2 enrollments. La user n'est testé que dans la méthode `export` car on a besoin de la notion de user pour tester le cas du login. 

> Pourtant il est présent dans enrollments, qui est présent dans spreadsheet_generator, qui est utilisé dans le it

je suis d'accord mais dans les fait avec le `context`, le `let` n'initialise pas l'enrollment dans chaque bloc alors qu'avec le `let!`, c'est le cas. 
Vu que le `context` n'a plus de sens, je le retire et le `let!` n'a plus de raison d'être.

Reply:
On verra ce que ça rend à la fin dtf.